### ISSUE TITLE
PS5 controller not working with Steam Input with `hid_playstation` module loaded

### ISSUE THREAD

**Issue by junglerobba** [ValveSoftware/steam-for-linux#7791]
#### Your system information

* Steam client version (build number or date): May 7 2021, 23:55:52
* Distribution (e.g. Ubuntu): Arch Linux
* Opted into Steam client beta?: [Yes/No] Yes
* Have you checked for system updates?: [Yes/No] Yes

#### Please describe your issue in as much detail as possible:
Describe what you _expected_ should happen and what _did_ happen. Please link any large code pastes as a [Github Gist](https://gist.github.com/)
After upgrading to Linux 5.12 the PS5 controller does not produce any inputs in games when used with Steam Input.
Running `rmmod hid_playstation` makes it work fine again.

#### Steps for reproducing this issue:

1. Plug in controller
2. Start Steam with PlayStation Steam Input enabled
3. Start game -> Controller produces no input

**Comment by hopibel** [comment]
Seems highly dependent on the game's input handling (tested with DualSense connected via bluetooth):

* Hollow Knight: processes input from both. West and North button inputs from `hid_playstation` are swapped (old SDL version?), Steam Input maps them correctly. This causes the game to receive both X and Y from the west button. Disabling either the driver or Steam Input fixes the double input problem though the poor native support means it's only playable with Steam Input.
* Celeste: works even with both the driver and Steam Input disabled. Looks like has native DirectInput support.
* Slay the Spire: ignores Steam Input, even when `hid_playstation` is unloaded. I was able to force it to use Steam Input by running the Windows version in Proton.
* CrossCode: unplayable. It's an NW app, so it uses chromium's gamepad API which doesn't map the triggers and joysticks correctly.

**Comment by TheTee82** [comment]
- Steam client version (build number or date): Jun 25 2021, at 18:26:27
- Steam API: v020
- Steam package versions: 1624650462
- Distribution (e.g. Ubuntu): Manjaro Linux
- Opted into Steam client beta?: [Yes/No] Yes
- Have you checked for system updates?: [Yes/No] Yes
- Kernel: `5.12.9-1-MANJARO x86_64`

---

I'm having similar issues here.
- when `hid_playstation` is enabled, no input whatsoever is working - except big picture navigation.
- when it's disabled, controllers seem to respond ingame when using Playstation Steam Input 
  - however the mapping to most games is completely wrong even when using default `Gamepad` controller mappings
- tested with both games running via Proton (eg. Yakuza: Like a Dragon) and native games (eg. Assault Android Cactus)
- tested with both bluetooth and tethered DualSense
- Steam seems to recognize and load the controller config however when - at least when starting a game on steam in terminal I'm getting this message (Yakuza: Like a Dragon):
```
Loaded Config for Local Selection Path for App ID 1235140, Controller 0: /home/myuser/.local/share/Steam//controller_base/templates/controller_ps5_gamepad_joystick.vdf
```
On Windows 10 the same controller works just fine with the same games - however as of right now, DualSense Controller on Linux is basically unusable for me.

**Comment by cyberang3l** [comment]
Same issue here with a bluetooth paired PS5 controller and Linux kernel 5.12.
Mainly tested with Hollow Knight.

When the `hid_playstation` module is loaded, I experience exactly what is described in this 4 years old bug with Hollow Knight: https://steamcommunity.com/app/367520/discussions/0/2592234299560251689/

When I `sudo rmmod hid_playstation`, the controller works fine.

**Comment by Vash63** [comment]
With Yakuza Kiwami 2 on 5.14 it works but uses Xbox button prompts, blacklisting hid_playstation restored the button prompts for me.

**Comment by z0z0z** [comment]
For me personally, I just need to connect the PS5 controller before starting games and then it freezes with a blank screen for like 1 minute before Steam gives me the "controller connected" notification in the overlay

Opening "controller configuration" in the Steam Overlay may help btw.

**Comment by Vash63** [comment]
> For me personally, I just need to connect the PS5 controller before starting games and then it freezes with a blank screen for like 1 minute before Steam gives me the "controller connected" notification in the overlay
> 
> Opening "controller configuration" in the Steam Overlay may help btw.

Yes I have the same here with every game I tested before I rebuilt my kernel without the hid_playstation module. I think Steam Input is not fully compatible with hid_playstation and they conflict.
